src/smc-webapp/editor_chat.cjsx

```diff
@@ -178,6 +178,11 @@ class ChatActions extends Actions
 
         if messages_before != messages
             @setState(messages: messages)
+            store = @redux.getStore(@name)
+            if store.get('saved_position') + store.get('offset') + 20 > store.get('height') or store.get('height') == store.get('client_height')
```

Comment:
This condition is hard to understand. What is the 20 for? What's the idea here? Maybe it should be a function in the store?